File: typescript/src/PlanetTS1/Model/Neutral/Buildings/NRGPlant.ts


abstract class NRGPlant extends Building {
    minCost : number
    gasCost : number
    nrgSal : number
    nrgCost : number
    commands : Array<string>
    health : number

    constructor() {
        super()
        this.minCost = 200
        this.gasCost = 100
        this.nrgSal = 5
        this.nrgCost = 0
        this.commands = new Array<string>("Abort")
        this.health = 100
    }
}

File: kotlinjs/old/NRGPlant.kt


abstract class NRGPlant() : Building() {
    override var minCost = 200
    override var gasCost = 100
    override var nrgSal = 5
    override var nrgCost = 0
    override var commands = mutableListOf<String>("Abort")
    override var health = 100
}